apps/obsidian/src/components/CreateNodeModal.tsx

```diff
@@ -0,0 +1,143 @@
+import { App, Modal, Notice } from "obsidian";
+import { DiscourseNode } from "~/types";
+import type DiscourseGraphPlugin from "~/index";
+
+interface CreateNodeModalProps {
+  nodeTypes: DiscourseNode[];
+  plugin: DiscourseGraphPlugin;
+  onNodeCreate: (nodeType: DiscourseNode, title: string) => Promise<void>;
+}
+
+export class CreateNodeModal extends Modal {
+  private nodeTypes: DiscourseNode[];
+  private plugin: DiscourseGraphPlugin;
+  private onNodeCreate: (
+    nodeType: DiscourseNode,
+    title: string,
+  ) => Promise<void>;
+  private selectedNodeType: DiscourseNode | null = null;
+  private titleInput: HTMLInputElement | null = null;
+  private confirmButton: HTMLButtonElement | null = null;
+
+  constructor(app: App, props: CreateNodeModalProps) {
+    super(app);
+    this.nodeTypes = props.nodeTypes;
+    this.plugin = props.plugin;
+    this.onNodeCreate = props.onNodeCreate;
+  }
+
+  onOpen() {
```

Comment:
One thing to considering here would be returning a react component.
This would allow us to start creating reusable components for common things (`Input`, `Button`, etc), use tailwind, and it would also be easier maintain (less verbose).

Can be deferred.